touchforms/formplayer/static/formplayer/script/xforgasm.js

```diff
@@ -16,8 +16,53 @@ var ERROR_MESSAGE = "Something unexpected went wrong on that request. " +
     "If you have problems filling in the rest of your form please submit an issue. " +
     "Technical Details: ";
 
+function TaskQueue() {
```

Comment:
A bit overkill for the task at hand, but I think we can extend this to refactor the whole BLOCKING_REQUESTS structure